Modules/_ctypes/callproc.c

```diff
@@ -1321,12 +1321,19 @@ static PyObject *format_error(PyObject *self, PyObject *args)
 {
     PyObject *result;
     wchar_t *lpMsgBuf;
-    DWORD code = 0;
-    if (!PyArg_ParseTuple(args, "|i:FormatError", &code))
+    long long code = 0;
+
+    if (!PyArg_ParseTuple(args, "|L:FormatError", &code))
+        return NULL;
+
+    if(code < LONG_MIN || code > ULONG_MAX) {
```

Comment:
```suggestion
    if((DWORD)code != code) {
```

Personally I prefer this test, as it's closer to what we're going to be doing with the value.